test/unit/math/opencl/prim/col_test.cpp

```diff
@@ -0,0 +1,26 @@
+#ifdef STAN_OPENCL
+#include <stan/math/opencl/prim/col.hpp>
+#include <stan/math/prim/fun/typedefs.hpp>
+#include <gtest/gtest.h>
+#include <vector>
+
+TEST(MathMatrixCL, col_exception) {
+  stan::math::matrix_cl<double> m1_cl(3, 3);
+  EXPECT_THROW(col(m1_cl, 0), std::invalid_argument);
+  EXPECT_THROW(col(m1_cl, 4), std::invalid_argument);
+}
+
+TEST(MathMatrixCL, col_value_check) {
+  stan::math::matrix_d m1(3, 3);
+  m1 << 1, 2, 3, 4, 5, 6, 7, 8, 9;
+
+  stan::math::matrix_cl<double> m1_cl(m1);
+  stan::math::matrix_cl<double> v_cl = stan::math::col(m1_cl, 3);
```

Comment:
Are we able to get the column of an expression like `stan::math::col(add(m1_cl, m1_cl), 3);`?

Reply:
Will add tests for those cases. Block works like that so this should too.